cosmwasm/contracts/crosschain-registry/src/execute.rs

```diff
@@ -40,6 +42,100 @@ pub struct ContractAliasInput {
     pub new_alias: Option<String>,
 }
 
+pub fn propose_pfm(
+    ctx: (DepsMut, Env, MessageInfo),
+    chain: String,
+) -> Result<Response, ContractError> {
+    let (deps, env, info) = ctx;
+
+    // enforce lowercase
+    let chain = chain.to_lowercase();
+
+    // validation
+    let registry = Registry::default(deps.as_ref());
+    let coin = cw_utils::one_coin(&info)?;
+    let native_chain = registry.get_native_chain(&coin.denom)?;
+
+    if native_chain.as_ref() != chain {
+        return Err(ContractError::CoinFronInvalidChain {
+            supplied_chain: native_chain.as_ref().to_string(),
+            expected_chain: chain,
+        });
+    }
+
+    // check if the chain is already registered or is in progress
+    if let Some(chain_pfm) = CHAIN_PFM_MAP.may_load(deps.storage, &chain)? {
+        if chain_pfm.is_validated() {
+            // Only authorized addresses can ask for a validated PFM to be re-checked
+            // If sender is the contract governor, then they are authorized to do do this to any chain
+            // Otherwise, they must be authorized to do manage the chain they are attempting to modify
+            let user_permission =
+                check_is_authorized(deps.as_ref(), info.sender, Some(chain.clone()))?;
+            check_action_permission(FullOperation::Change, user_permission)?;
+        } else {
+            return Err(ContractError::PFMValidationAlreadyInProgress {
+                chain: chain.clone(),
+            });
+        }
```

Comment:
In the else case we return `PFMValidationAlreadyInProgress`. Does not being validated guarantee that validation has started?

Reply:
It seems to me that this case represents as "have been proposed but not necessarily started validation" as validation is a separate msg execute

Reply:
I think it's just different semantics here. The way I was thinking about it is: when a PFM check is proposed, then validation is considered to be in progress (propose = a proposal to validate PFM on a chain).

Reply:
>  Does not being validated guarantee that validation has started?

Here it does, as we are calling `is_validated()` on a ChainPFM fetched from storage. If it is in storage, it means someone has proposed it. If validation fails, we remove it from storage